### PR TITLE
Add travis support

### PR DESCRIPTION
**Pull request by pbjd**
Add some additional make recipes for cleaning up some (optional) downloaded build dependencies.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+language: cpp
+script: 
+  - make init-submodule 
+  - make
+  - make check
+compiler: gcc
```
Comment (pb-cdunn) [inline .travis.yml]: It's possible to make this run for *both* gcc *and* clang. But we can add that later.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+before_install:
+- sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
+- sudo apt-get update -qq
+- sudo apt-get install -qq libyajl-dev libxml2-dev libxqilla-dev
+- if [ "$CXX" = "clang++" ]; then sudo apt-get install -qq libstdc++-4.8-dev; fi
```
Comment (pb-cdunn) [inline .travis.yml]: Why is 4.8 so important?

File: Makefile [hunk 3]
```diff
@@ -12,6 +12,9 @@ init-submodule:
 	git submodule update
 	$(MAKE) -C blasr_libcpp/alignment nohdf=1
 	$(MAKE) -C blasr_libcpp/pbdata
+
+submodule-clean:
+	$(RM) -r blasr_libcpp
```
Comment (pb-cdunn) [inline Makefile]: cd blasr_libcpp; git clean -xdf

Removing the submodule is ok, I guess, but not really necessary.

File: .travis.yml (hunk 2 above)
Comment (pbjd) [inline .travis.yml]: C++11 support (i.e., -std=c++11)

File: README.md [hunk 4]
```diff
@@ -1,3 +1,4 @@
+[![Build Status](https://travis-ci.org/pbjd/pbdagcon.svg?branch=master)](https://travis-ci.org/pbjd/pbdagcon)
```
Comment (pb-cdunn) [inline README.md]: I'd drop the branch name here.

And this confuses me. Recursive markup(markup)? What does this do?

File: Makefile (hunk 3 above)
Comment (pbjd) [inline Makefile]: I just added that as a convenience for testing out first-time builds.

File: README.md (hunk 4 above)
Comment (pbjd) [inline README.md]: Taken directly from: http://docs.travis-ci.com/user/status-images/

Just a way of embedding the status of the last travis build result into the README.  Useful indicator for devs.

Comment (pb-cdunn) [inline README.md]: Oh! The `!` embeds an image. Of course. Nice idea.

**Pull request merged**